src/kp/kp_client.py

```diff
@@ -120,14 +129,33 @@ def request_session_token(self):
             if cookie['name'] == 'JSESSIONID':
                 session_cookie = cookie['value']
                 break
+
         self.session_cookie = session_cookie
-        return session_cookie
+        self.isFetchingToken = False
```

Comment:
This will not be set if an exception occurs. Maybe wrap it all in a try/except and also add it to the json exception block

Reply:
No longer possible to my understanding: https://github.com/Randers-Kommune-Digitalisering/api-service/blob/29a0a2004d1f275f4be92fd4f8b6ac0160689993/src/kp/kp_client.py#L135

No except blocks return anything, thus self.isFetchingToken will always be set before returning from the function.

Reply:
But if an unforeseen exception occurs(not requests.exceptions.RequestException or requests.exceptions.JSONDecodeError), it will not be set and thus stuck

Reply:
Wrapped function in try except clause